lib/cocoapods-core/podfile/target_definition.rb

```diff
@@ -307,14 +309,21 @@ def inhibit_all_warnings=(flag)
 
       # Inhibits warnings for a specific pod during compilation.
       #
-      # @param  [String] pod name
-      #         Whether the warnings should be suppressed.
+      # @param  [String] pod_name
+      #         Name of the pod for which the warnings will be inhibited or not.
+      #
+      # @param  [Bool] should_inhibit
+      #         Whether the warnings should be inhibited or not for given pod.
       #
       # @return [void]
       #
-      def inhibit_warnings_for_pod(pod_name)
-        raw_inhibit_warnings_hash['for_pods'] ||= []
-        raw_inhibit_warnings_hash['for_pods'] << pod_name
+      def set_inhibit_warnings_for_pod(pod_name, should_inhibit)
+        hash_key = 'for_pods' if should_inhibit == true
```

Comment:
If I do this, pod will inhibit for the cases user sets the value any object that is not boolean false or nil, such as `:inhibit_warnings => 'false'`. So I think we should explicitly check like this and ignore otherwise.